### ISSUE TITLE
Wrong "Not Accessed" warning on class method starts with double underscores

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#713]
**Describe the bug**
When a class method name starts with double underscore, "Not accessed" warning appears if they are only accessed within the class.

Please see the example below.

![image](https://user-images.githubusercontent.com/3479453/83938524-9ff39680-a807-11ea-96ff-6841fe5f997a.png)

**To Reproduce**
```python
class A:
    def some_public_method(self):
        self.__some_private_method()

    def __some_private_method(self):
        pass
```

**Expected behavior**
Show no "not accessed" warnings for such methods because they are acturaly accessed.

**VS Code extension or command-line**
VSCode extension 1.1.40

**Additional context**
Please tell me this is not an expected behavior because we have a naming conventions says "Use double underscore prefix for private methods" 😸

**Comment by erictraut** [comment]
I'm not able to repro the problem as reported.

I copied your code exactly and added a `# pyright: reportUnusedFunction=true` comment at the top of the file to enable this rule. I don't see any error. If I then replace the invocation of the private method with `pass`, I see the error as expected.

![Screen Shot 2020-06-06 at 12 37 09 AM](https://user-images.githubusercontent.com/7040122/83938966-e0163300-a78d-11ea-92ab-7775eb52deca.png)

Is there perhaps something more that I need to do to repro the problem?

**Comment by giyyapan** [comment]
🤦 
Emmm... I was ... not able to repro this myself anymore!

When I see your comment, I tried to create a new demo project to repro this, and I disabled all extension for that. I enabled only Python and Pyright after that, but the issue can't be reproduced in the new project.

Then I went back to my main project, enabled all extension I used to use, and there's no "not accessed" warning anymore!

I don't know what exactly happend, but disabling all extension (maybe just Pyright?) seems to fixed it. I think it might related to local cache or something, but it's good to see it gone.

Well, I think I'll just close this issue for now and see if I (or my colleague) can repro this constantly. Thanks for your swift response Eric!
